os/kernel/pthread/pthread_mutexunlock.c

```diff
@@ -117,16 +146,24 @@
 
 int pthread_mutex_unlock(FAR pthread_mutex_t *mutex)
 {
-	int ret = EINVAL;
+	int ret = OK;
 
 	svdbg("mutex=0x%p\n", mutex);
 	DEBUGASSERT(mutex != NULL);
+	if (mutex == NULL) {
+		return EINVAL;
+	}
 
 	/* Make sure the semaphore is stable while we make the following checks.
 	 * This all needs to be one atomic action.
 	 */
 	sched_lock();
-	if (mutex != NULL) {
+
+	/* The unlock operation is only performed if the mutex is actually locked.
+	 * If the mutex is not locked, then SUCCESS will be returned (there is
+	 * no error return value specified for this case).
+	 */
+	if (pthread_mutex_islocked(mutex)) {
```

Comment:
Ok. I suggest to set ret value as EINVAL,  because trying to unlock an unlocked mutex is an invalid attempt .    And also we can make use of the  pid  mutex->pid != -1 ,  it means, mutex is already locked and thus avoid function call and improve performance. however you can take a decision. Thanks, Pradeep